xmtp_mls/src/subscriptions/stream_all.rs

```diff
@@ -409,4 +412,102 @@ mod tests {
         tracing::info!("Total Messages: {}", messages.len());
         assert_eq!(messages.len(), 5);
     }
+
+    #[wasm_bindgen_test(unsupported = tokio::test(flavor = "multi_thread", worker_threads = 5))]
+    async fn test_stream_all_messages_with_metadata_update() {
+        // Create test clients
+        let alix = ClientBuilder::new_test_client(&generate_local_wallet()).await;
+        let bo = ClientBuilder::new_test_client(&generate_local_wallet()).await;
+
+        let alix_group = alix
+            .create_group(None, GroupMetadataOptions::default())
+            .unwrap();
+
+        // Start Bo's message stream
+        let stream = bo.stream_all_messages(None).await.unwrap();
+        futures::pin_mut!(stream);
+
+        // IF WE MOVE THIS LINE ABOVE THE STREAM, THE TEST PASSES
+        alix_group
```

Comment:
@insipx @codabrink 

See new test in `xmtp_mls` that reproduces the issue. I noticed that it passes if you move alix inviting bo to the group before the stream starts.